examples/environments/auction_simulation/main.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+"""An auction simulation."""
+import argparse
+
+from agents import Item, Auctioneer, Bidder, RandomBidder
+from env import Auction
+from listeners import StartListener, BidListener, BidTimerListener
+
+import agentscope
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse arguments"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--bidder-num", type=int, default=5)
+    parser.add_argument(
+        "--agent-type",
+        choices=["random", "llm"],
+        default="random",
+    )
+    parser.add_argument("--waiting-time", type=float, default=3.0)
+    return parser.parse_args()
+
+
+def main(
+    bidder_num: int = 5,
+    agent_type: str = "random",
+    waiting_time: float = 3.0,
+) -> None:
+    """The main function."""
+    agentscope.init(
+        project="auction_simulation",
+        name="main",
+        save_code=False,
+        save_api_invoke=False,
+        model_configs="configs/model_configs.json",
+        use_monitor=False,
+    )
+
+    auction = Auction("auction")
+
+    auctioneer = Auctioneer("auctioneer", auction, waiting_time=waiting_time)
+    if agent_type == "random":
```

Comment:
Bidders can be a part of the `Auction`. 

For examples, add `register_bider`, `remove_bider` 
 and `get_biders` to `Auction`, then the `StartListener` can get all bidders from env.
In this case, only one `StartListener` is needed.

Reply:
If getting all bidders in `StartListener`, I have to manually write a thread pool in the `StartListener` to concurrently ask each bidder whether they want to bid on the item, instead of using the concurrency in the `trigger_listener`. Maybe it seems a bit strange?